tests/Temporalio.Tests/AssertMore.cs

```diff
@@ -127,11 +128,21 @@ public static void Every<T>(IEnumerable<T> items, params Action<T>[] actions)
                         found = true;
                         break;
                     }
-                    catch (Xunit.Sdk.XunitException)
+                    catch (Xunit.Sdk.XunitException ex)
+                    {
+                        exceptions.Add(ex);
+                    }
+                }
+                if (!found)
+                {
+                    Console.WriteLine($"Failed to match item: {item}");
+                    Console.WriteLine("Attempted matches and their failures:");
+                    for (int i = 0; i < exceptions.Count; i++)
                     {
+                        Console.WriteLine($"  Match {i + 1}: {exceptions[i].Message}");
                     }
+                    Assert.True(found, $"Item {item} had no match");
                 }
```

Comment:
Would rather not write to stdout in this utility, if you really need to give back the exceptions, can probably change to:

```suggestion
                if (!found)
                {
                    throw new AggregateException($"Item {item} had no match", exceptions);
                }
```